gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/DistributedRegion.java

```diff
@@ -4865,7 +4865,9 @@ void updateSizeOnRemove(Object key, int oldSize) {
   @Override
   void updateSizeOnClearRegion(int sizeBeforeClear){
     if(!this.reservedTable() && needAccounting()){
-      callback.dropStorageMemory(getFullPath(), getIgnoreBytes());
+      long ignoreBytes = this.isDestroyed() ? getIgnoreBytes() :
+              getIgnoreBytes() + regionOverHead;
+      callback.dropStorageMemory(getFullPath(), ignoreBytes);
```

Comment:
My point was if both BucketRegion & DistributedRegion had the same code we could have refactored it into LocalRegion. Now as both have slightly different logic, it's better to keep them separate but consistent in terms of method names.